SU-SOE/stanford_bean_types
# SOE-3047: initial commit for adding superhead to testimonials.

# READY FOR REVIEW

# Summary
- This updates the Testimonial block by adding the superhead field and a new display view mode.
So far we haven't used the Testimonial block on this site.

# Needed By (Date)
- Soon! since we'll be adding another PR to this repo. 😄 

# Criticality
- Client is eagerly awaiting this.


# Steps to Test
## Update feature
- Switch to this branch - 7.x-2.x-SOE-3047-testimonial
- Revert the feature:
`drush fr stanford_bean_types --force -y`
- Clear cache
`drush cc all`
## Update and verify through UI
- Go to `/admin/structure/block-types/manage/stanford-testimonial-block/display`
- In the vertical tabs at the bottom click on **Custom Display Settings**
- Click on **SoE Default**
![screen shot 2018-06-06 at 16 02 45](https://user-images.githubusercontent.com/284440/41069538-30e03a22-69a3-11e8-83ef-ec5108817564.png)

- Click **Save**
- After the page refreshes click on the **SoE Default** button in the top right corner and verify:
![screen shot 2018-06-06 at 15 59 26](https://user-images.githubusercontent.com/284440/41069422-aa07d71c-69a2-11e8-945e-a1f2b3502567.png)

## Test create & display
- Create a new testimonial block and fill out the fields:
```
Superhead
Quote
Credit
View Mode = SoE Default
```
- Place block on a page and verify it looks like this _without_ the teal colored bar:
(The colored bar will go into the CSS in SoE Helper)
![superhead_testimonial 4](https://user-images.githubusercontent.com/284440/41067428-69ef0bee-699a-11e8-818f-231dfa25f4aa.jpg)

# Affects 
- SoE (School of Engineering website)

# Associated Issues and/or People
## Related JIRA ticket(s)
work: https://stanfordits.atlassian.net/browse/SOE-3047
PR:  https://stanfordits.atlassian.net/browse/SOE-3081

## Related PRs

## More Information

## Folks to notify
@kerri-augenstein 



# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
